handlers/handlers.go

```diff
@@ -246,8 +249,51 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(notesInJson))
 
+	case http.MethodPost:
+		type NoteForm struct {
+			Content  string `json:"content"`
+			Category string `json:"category"`
+		}
+
+		noteForm := new(NoteForm)
+
+		if err := json.NewDecoder(request.Body).Decode(noteForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(strings.TrimSpace(noteForm.Content)) == 0 {
+			http.Error(responseWriter, "Note content cannot be empty or just whitespace", http.StatusBadRequest)
+			return
+		}
+
+		note := models.CreateNewNote(userId, noteForm.Content)
```

Comment:
this confused me when i read it. Do we actually need this function? Why not just create the object here?

Reply:
What if we call it `InstantiateNewNote`? Its my OOP-ness that makes me always want to create initializes, since its more modular. However, since its only used once here. I don't mind inlining it.

Reply:
cool yeah, let's inline it for now and we can talk about creating a "constructor" if we end up writing this once or twice more